File: cltch-ntwrk/native-ios-swift/CLTCHNTWRKSwift/Models/UserSession.swift
import Foundation

struct UserSession: Identifiable, Codable {
    let id: String
    let email: String
    let displayName: String
    let role: UserRole
    let city: String
    let state: String
    let completionPercent: Int
    let status: String
    let upcomingBookings: [Booking]
    let metrics: [DashboardMetric]
    let spotlightActions: [WorkspaceAction]
    let radarMatches: [Booking]
    let profileFacts: [ProfileFact]
    let supportItems: [SupportItem]
}

struct DashboardMetric: Identifiable, Codable {
    let id: String
    let title: String
    let value: String
    let accentKey: String
}

struct WorkspaceAction: Identifiable, Codable {
    let id: String
    let title: String
    let detail: String
}

struct ProfileFact: Identifiable, Codable {
    let id: String
    let label: String
    let value: String
}

struct SupportItem: Identifiable, Codable {
    let id: String
    let title: String
    let detail: String
    let status: String
}

File: cltch-ntwrk/native-android-kotlin/app/src/main/java/com/cltch/ntwrk/nativeapp/model/UserSession.kt
package com.cltch.ntwrk.nativeapp.model

data class UserSession(
    val id: String,
    val email: String,
    val displayName: String,
    val role: UserRole,
    val city: String,
    val state: String,
    val completionPercent: Int,
    val status: String,
    val bookings: List<Booking>,
    val metrics: List<DashboardMetric>,
    val spotlightActions: List<WorkspaceAction>,
    val radarMatches: List<Booking>,
    val profileFacts: List<ProfileFact>,
    val supportItems: List<SupportItem>
)

data class DashboardMetric(
    val id: String,
    val title: String,
    val value: String,
    val accentKey: String
)

data class WorkspaceAction(
    val id: String,
    val title: String,
    val detail: String
)

data class ProfileFact(
    val id: String,
    val label: String,
    val value: String
)

data class SupportItem(
    val id: String,
    val title: String,
    val detail: String,
    val status: String
)
